homeassistant/components/light/__init__.py

```diff
@@ -609,9 +623,16 @@ def capability_attributes(self):
         if supported_features & SUPPORT_EFFECT:
             data[ATTR_EFFECT_LIST] = self.effect_list
 
-        data[ATTR_SUPPORTED_COLOR_MODES] = sorted(
-            self._light_internal_supported_color_modes
-        )
+        supported_color_modes = self._light_internal_supported_color_modes
+        try:
+            supported_color_modes = validate_supported_color_modes(
```

Comment:
We don't validate attributes at runtime as it can get too costly to do it for everything. Instead I once had this idea #37663

Reply:
In this particular case, it's a capability attribute which should rarely change, if ever.
If the attribute has not changed, there's no need to re-validate.

In #37663 it's mentioned that:
> It's up to each integration to implement a validator platform.

So it's opt-in per integration? Can't it be done in base components instead?